src/dbos-executor.ts

```diff
@@ -365,7 +370,7 @@ export class DBOSExecutor {
     }
     const wConfig = wInfo.config;
 
-    const wCtxt: WorkflowContextImpl = new WorkflowContextImpl(this, params.parentCtx, workflowUUID, wConfig, wf.name, presetUUID);
+    const wCtxt: WorkflowContextImpl = new WorkflowContextImpl(this, params.parentCtx, workflowUUID, wConfig, wf.name, presetUUID, params.tempWfType, params.tempWfName);
```

Comment:
Previously, we set the temporary workflow name and type inside of a temporary workflow. However, we need to know their name before running them, so we have to pass the name and type into the workflow context.